bootstrap/terraform/scripts/imds_access_token.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# External data source script for obtaining an access token JWT from the Azure 
+# Instance Metadata Service (IMDS) endpoint. The access token is used as the form
+# of authentication for the Vault Azure Auth Method. For details, see
+# - https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http
+#- https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/data_source
+# - https://developer.hashicorp.com/vault/api-docs/auth/azure#jwt
```

Comment:
This seems to work well, but there could be a better way to do this. I'm not super Terraform savvy. Open to ideas!